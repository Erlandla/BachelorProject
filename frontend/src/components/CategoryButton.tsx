import React from 'react'
import PropTypes from 'prop-types'

function CategoryButton({
  text,
  focused,
  onClick,
}: {
  text: string
  focused: string
  onClick: any
}) {
  const defaultStyle =
    'bg-buttonDark border-4 border-buttonDark text-white cursor-pointer px-5 py-1 rounded-[45px] h-10 w-200px hover:bg-buttonHover hover:border-buttonHover'
  const focusedStyle =
    'bg-buttonDark border-4 border-textLight text-white cursor-pointer px-5 py-1 rounded-[45px] h-10 w-200px'

  return (
    <div
      onClick={() => onClick(text)}
      className={text === focused ? focusedStyle : defaultStyle}
      data-testid="categoryButton"
    >
      <p className="whitespace-nowrap" data-testid="categoryText">
        {' '}
        {text}
      </p>
    </div>
  )
}

CategoryButton.propTypes = {
  onClick: PropTypes.func.isRequired,
}

export default CategoryButton
