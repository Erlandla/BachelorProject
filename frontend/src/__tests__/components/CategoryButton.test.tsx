import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import CategoryButton from "../../components/CategoryButton";
import renderer from "react-test-renderer";

afterEach(() => {
    cleanup();
});

it("Component renders correctly", () => {
    render(
        <CategoryButton 
            text="This is test category :^)" 
            focused="Yes" 
            onClick={() => {}}
        />
    );

    const categoryText = screen.getByTestId("categoryText");
    expect(categoryText).toBeInTheDocument();
});

it("Component reacts to click events", () => {
    let testVariable: number = 0;

    render(
        <CategoryButton 
            text="This is test category :^)" 
            focused="Yes" 
            onClick={() => { testVariable = 1; }}
        />
    );
    const button = screen.getByTestId("categoryButton");
    button.click();
    expect(testVariable).toEqual(1);
});

it("Matches snapshot", () => {
    const snapshot = renderer.create(<CategoryButton
        text="Oh snap!"
        focused="I guess"
        onClick={() => {}}
    />).toJSON();
    expect(snapshot).toMatchSnapshot();
});
