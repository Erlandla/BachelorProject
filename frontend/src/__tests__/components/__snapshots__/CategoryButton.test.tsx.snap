// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="bg-buttonDark border-4 border-buttonDark text-white cursor-pointer px-5 py-1 rounded-[45px] h-10 w-200px hover:bg-buttonHover hover:border-buttonHover"
  data-testid="categoryButton"
  onClick={[Function]}
>
  <p
    className="whitespace-nowrap"
    data-testid="categoryText"
  >
     
    Oh snap!
  </p>
</div>
`;
